docs/source/contributing/layout.rst

```diff
@@ -0,0 +1,126 @@
+===============================
+
+This page covers the layout and structure of the JupyterHub repository. It provides brief information on the purpose of the various files and directories in the repository. Please consult JupyterHub documentation for more details.
+
+
+Directory Structure
+===================
+
+ci/
+    This directory contains files for automating aspects of the continuous integration pipeline of the JupterHub project; some of which includes checking distribution packages to ensure they contain all expected files.
+
+demo-image/
+    This directory contains a Dockerfile for building a JupyterHub image meant for demo or testing purposes only. You can check out the `README` file in the directory for more details.
+
+dockerfiles/
+    This directory contains the Dockerfile for building the JupyterHub base image. You can check out the `README` in the directory for more details.
+
+docs/
+    This directory contains the documentation files for the JupyterHub project.
+
+examples/
+    This directory details several services provided by the JupyterHub project, alongside their configurations to enable you configure JupyterHub to use those services.
+
+jsx/
+    This directory contains current updates to the JupyterHub Admin Dashboard.
+
+JupyterHub/
+    This directory contains the core code files for the JupyterHub project.
+
+onbuild/
+    This directory contains the Dockerfile for building a JupyterHub image. Specifically, it would include the configuration file(`JupyterHub_config.py`) in the image build.
+
+share/jupyterHub/
+    This directory contains JupyterHub data and static files.
+
+singleuser/
+    This directory contains the Dockerfile for building a JupyterHub image for a single user. You can check out the `README` in the directory for more details.
+
+testing/
+    This directory contains a sample JupyterHub configuration file for testing.
+
+Executable files
+================
+
+setup.py
+    The packaging file for the JupyterHub codebase. It is used in building the JupyterHub distribution package. It contains information on the name, version, data files, package files, python version requirement, entry points, and the dependencies of the JupyterHub codebase. 
+
+Dockerfile
+    The Dockerfile containing the base image for running JupyterHub.
+
+bower-lite
+    Another components file used in packaging JupyterHub. It stages the frontend dependencies of the JupyterHub repository from the node_modules directory into the components sub-directory. Check the static directory in the `share/jupyterHub` directory to find the components sub-directory.
+
+package.json
+    This file contains information on JupyterHub nodejs dependencies.
+
+pytest.ini
+    A configuration file for customizing Pytest behaviors to suit the usage of Pytest in the JupyterHub project. Pytest is a python tool for running tests.
+
+pyproject.toml
+    A requirement for using setuptools(a build backend) in building the JupyterHub distribution package.
```

Comment:
pyproject.toml is actually used to configure a few tools. Specifically, code formatting and release tools (tbump).